### ISSUE TITLE
Audacity 3.7.1 on Apple M1 does not recognize ffmpeg 6.1 or 7.1 library, 3.7.0 recognized just 6.1

### ISSUE THREAD

**Issue by sidney** [audacity/audacity#7966]
### Bug description

Audacity 3.7.0 on Apple silicon did not recognize ffmpeg libraries that used the version 7 API, though it had a workaround of brew install ffmpeg@6 to get ffmpeg 6.1.2 libraries. However, Audacity 3.7.1 no longer recognizes the ffmpeg 6.1.2 library, nor does it recognize the current ffmpeg version 7.1.2 library. I did not try building ffmpeg 7.0.2 from source to see if Audacity 3.7.1 recognizes it.

### Steps to reproduce

I tested on a Macbook Pro M1 running Sonoma, but expect it is true of any Apple silicon
1. With Homebrew installed, run brew install ffmpeg and brew install ffmpeg@6 to install those two packages
2. Install Audacity version 3.7.0
3. In Audacity Preferences | Libraries click the FFmpeg Library Locate button
4. Press command-shift-. to display hidden files and directories in file browser dialog
5. Navigate to /opt/homebrew/Celler/ffmpeg/6.1.2/lib/libavformat.dylib and select it
6. Verify that Audacity does recognize that library
7. Navigate to /opt/homebrew/Celler/ffmpeg/7.1*/lib/libavformat.dylib (The * will depend on your installation)
8. Verify that Audacity does not recognize that library you have selected
9. Quit Audacity, then install version 3.7.1
10. repeat steps 3 through 8, but verify that Audacity no longer recognizes ffmpeg 6.1.2 in step 6

### Expected behavior

When using Audacity 3.7.1 it would be best if it recognized the ffmpeg 7.1* library, but if it cannot, it shouild at least recgnize the ffmpeg 6.1.2 library so it could be used as a workaround to the lack of support for ffmpeg 7.1

### Actual behavior

Audacity 3.7.1 does not recognize the library files built for either version 6.1.2 or 7.1 of ffmpeg.

### Audacity Version

Audacity 3.7.x

### Operating system

macOS

### Additional context

MacOS on Apple Silicon, ffmpeg installed using brew install ffmpeg@6 and brew install ffmpeg
No attempt was made to install versions of ffmpeg that require building from source on Mac, only those two released versions that are available for installation from package binaries.

**Comment by dozzzzer** [comment]
I'm on MacOS 14.5, and Audacity 3.7.1 picks up ffmpeg 7.1 that I've installed using `brew install ffmpeg`.

<img width="600" alt="image" src="https://github.com/user-attachments/assets/8dfd46b4-01a6-47c5-be0a-66834127ff2f" />

<img width="600" alt="image" src="https://github.com/user-attachments/assets/5403981e-6436-42e0-91b1-8d4ef04998a0" />

**Comment by sidney** [comment]
I reinstalled ffmpeg and Audacity 3.7.1 and it works now. Sorry for the false alarm. Closing this issue.

**Comment by dozzzzer** [comment]
No problem :) @sidney

**Comment by ghost** [comment]
**For people getting here from Google:** The [Installing FFmpeg > Troubleshooting](https://support.audacityteam.org/basics/installing-ffmpeg#troubleshooting) section mentions that `ffmpeg` might have been auto-updated to a not yet supported version.

This was the case for me, so I installed  `ffmpeg@7` to work with Audacity 3.7.5 (path to the lib should be `/opt/homebrew/Cellar/ffmpeg@7/7.1.2/lib/libavformat.dylib`).

**Comment by Waitsnake** [comment]
> **For people getting here from Google:** The [Installing FFmpeg > Troubleshooting](https://support.audacityteam.org/basics/installing-ffmpeg#troubleshooting) section mentions that `ffmpeg` might have been auto-updated to a not yet supported version.
> 
> This was the case for me, so I installed `ffmpeg@7` to work with Audacity 3.7.5 (path to the lib should be `/opt/homebrew/Cellar/ffmpeg@7/7.1.2/lib/libavformat.dylib`).

Oh thanks for this tip. I looked quite a while what the cause of the problem was and how to solve it since Audacity UI gives you not even an error message. I had the same problem that my homebrew upgraded libavformat version 8.0.1 maybe a while ago (I update homebrew regulary but not use Audacity with the same rate) and ffmpeg was no longer kompatibel with Audacity, even after I updated to newest version of Audacity 3.7.5.

Doing a "brew reinstall ffmpeg@7" did the trick and get it back working and I have now the 8.x and 7.x versions installed.

Since you not get a error message from Audacity what the problem is or that there is a problem at all (it says nothing even after selecting it and looks like before selection) and it can be quite confusing from an UI usability point of view.
